.github/workflows/maven.yml

```diff
@@ -24,3 +24,30 @@ jobs:
         cache: maven
     - name: Build with Maven
       run: mvn -B test --file pom.xml
+    - name: Generate JaCoCo Badge
+      id: jacoco
+      uses: cicirello/jacoco-badge-generator@v2
+      with:
+          generate-branches-badge: true
+          generate-summary: true
+          fail-if-coverage-less-than: 50
+          fail-on-coverage-decrease: true
+
+    - name: Log coverage percentage(if it changed)
+      if: ${{ github.event_name == 'push' }}
+      run: |
+          echo "coverage = ${{ steps.jacoco.outputs.coverage }}"
+          echo "branch coverage = ${{ steps.jacoco.outputs.branches }}"
+    - name: Commit and push the badge (if it changed)
+      if: ${{ github.event_name == 'push' }}
+      uses: EndBug/add-and-commit@v7
+      with:
+        default_author: github_actions
+        message: 'commit badge'
+        add: '*.svg'
+    - name: Upload JaCoCo coverage report
+      if: ${{ github.event_name == 'push' }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: jacoco-report
+        path: target/site/jacoco/
```

Comment:
Shouldn't this be in a file of its own? Called jacoco.yml, code_coverage.yml or something similar? Otherwise it will be bundled up with the maven build action.